include/triton/core/tritonbackend.h

```diff
@@ -1411,6 +1411,36 @@ TRITONBACKEND_ISPEC TRITONSERVER_Error* TRITONBACKEND_GetBackendAttribute(
     TRITONBACKEND_Backend* backend,
     TRITONBACKEND_BackendAttribute* backend_attributes);
 
+/// Query the total memory usage of the model and its associated model
```

Comment:
Instead of Triton Core calling this function, I was wondering whether we can delegate this responsibility to the backend?

The backend can call `TRITONBACKEND_ReportModelMemoryUsage` whenever it wants and we would update the usage statistics for this model. For example, this would allow the backend to update the usage during actual execution if it notices any memory changes.

Reply:
This is a good idea

Reply:
I was thinking the same based on other thread of back and forth with core and num_usage==-1 🚀